### PR TITLE
Video: use youtube-nocookie.com for embedded videos

### PR DESCRIPTION
**Pull request by breville**
This converts the centralized dashboard & pegasus video players, along with a variety of other embedded videos, to use https://youtube-nocookie.com instead of https://youtube.com.

Data collection done before this change was set up in https://github.com/code-dot-org/code-dot-org/pull/21885.  We tested the new domain on a couple videos successfully with https://github.com/code-dot-org/code-dot-org/pull/22177.

Note that we sometimes rely on access to https://www.youtube.com/iframe_api for https://youtube-nocookie.com videos.  In cases where youtube.com is inaccessible, the video still appears to play, but doesn't auto-close its modal at the end.  Auto-closing was the feature added in https://github.com/code-dot-org/code-dot-org/pull/5633.

### CODE REVIEW

File: dashboard/app/models/video.rb [hunk 1]
```diff
@@ -24,9 +24,9 @@ class Video < ActiveRecord::Base
   # YouTube video IDs must be 11 characters and contain no invalid characters, such as exclamation points or asterisks.
   # Ref: https://developers.google.com/youtube/iframe_api_reference (events|onError|2)
   YOUTUBE_ID_REGEX = /[^!*"&?\/ ]{11}/
-  # YouTube embed URL has the following format: http://www.youtube.com/embed/VIDEO_ID
+  # YouTube embed URL has the following format: http://www.youtube-nocookie.com/embed/VIDEO_ID
   # Ref: https://developers.google.com/youtube/player_parameters#Manual_IFrame_Embeds
-  EMBED_URL_REGEX = /(?:http[s]?:)?\/\/(?:www\.)?(?:youtube(?:education)?)\.com\/embed\/(?<id>#{YOUTUBE_ID_REGEX})/
+  EMBED_URL_REGEX = /(?:http[s]?:)?\/\/(?:www\.)?(?:youtube(?:education)?(-nocookie)?)\.com\/embed\/(?<id>#{YOUTUBE_ID_REGEX})/
```
Comment (Hamms) [inline dashboard/app/models/video.rb]: Is youtubeeducation-nocookie a thing?

Comment (breville) [reply dashboard/app/models/video.rb]: Good catch; it isn't.  I think 
```
(?:http[s]?:)?\/\/(?:www\.)?(?:youtube(education|-nocookie)?)\.com\/embed
```
is better.

Comment (breville) [reply dashboard/app/models/video.rb]: Or should I make sure it's a non-capturing group like the others, using `?:`?

```
(?:http[s]?:)?\/\/(?:www\.)?(?:youtube(?:(education|-nocookie))?)\.com\/embed
```

Comment (breville) [reply dashboard/app/models/video.rb]: Actually after playing more with rubular.com, looks like this is it:

```
(?:http[s]?:)?\/\/(?:www\.)?(?:youtube(?:education|-nocookie)?)\.com\/embed
```

Given how this is [consumed](https://github.com/code-dot-org/code-dot-org/blob/1aca739c7b7be175ca5e54d219da56b45e91e3d5/dashboard/config/initializers/markdown_handler.rb#L17), making it a non-capturing group seems important.

**Comment by breville** [comment]
Good question.  Auto-close should only fail when `youtube.com` is blocked but `youtube-nocookie.com` isn't, which will certainly happen for a small percentage of real visitors, but shouldn't happen on our test environment.

**Comment by breville** [comment]
Oops, I merged this early, forgetting that we have a communication plan around making this change.  I will revert shortly.

**Pull request merged**